barretenberg/cpp/src/barretenberg/op_queue/ecc_ops_table.hpp

```diff
@@ -270,7 +273,52 @@ class UltraEccOpsTable {
         }
     }
 
-    std::vector<UltraOp> get_reconstructed() const { return table.get_reconstructed(); }
+    size_t get_unmerged_subtable_size() const { return table.get_unmerged_subtable_size(); }
+
+    std::vector<UltraOp> get_reconstructed() const
+    {
+        if (has_fixed_append && fixed_append_offset.has_value()) {
+            return get_reconstructed_with_fixed_append();
+        }
+        return table.get_reconstructed();
+    }
+    std::vector<UltraOp> get_reconstructed_with_fixed_append() const
+    {
+
+        ASSERT(get_unmerged_subtable_size() == 0,
+               "current subtable should be merged before reconstructing the full table of operations.");
+
+        std::vector<UltraOp> reconstructed_table;
+        // reconstructed_table.reserve(reconstructed_table_size);
+
+        for (size_t subtable_idx = 0; subtable_idx < table.num_subtables() - 1; subtable_idx++) {
+            const auto& subtable = table.get()[subtable_idx];
+            for (const auto& op : subtable) {
+                reconstructed_table.push_back(op);
+            }
+        }
+
+        // Add zeros if fixed offset is larger than current size
+        if (has_fixed_append && fixed_append_offset.has_value()) {
+            size_t current_size = reconstructed_table.size();
+            size_t target_offset = fixed_append_offset.value();
+
+            // Fill gap with no-ops if needed
+            while (current_size < target_offset) {
```

Comment:
FYI could just do e.g.
```
if (reconstructed_table.size() < target_offset) {
    reconstructed_table.resize(target_offset, UltraOp{/*no-op*/});
}
```

Reply:
But actually would probably be best to `reserve` the full size for `reconstructed_table` upfront then do something like `reconstructed_table.insert(reconstructed_table.end(), target_offset-current_size, UltraOp{});`

Reply:
these are great suggestions thanks!